### PR TITLE
Try/import css

### PR DESCRIPTION
**Pull request by yurynix**
Attempt at #11959:

Import CSS (like any other module) directly into our component modules and inject the styles into the page when the component is mounted

#### Testing:
1. `node bin/refactor-to-import-sass.js`
2. `make run`
3. To start over: `rm client/exclude-scss-from-build && git checkout client/* && git checkout assets/*`

#### How it works:
`bin/refactor-to-import-sass.js` traverses all directories in `client/`, it looks for scss files.
In directories it finds scss files, it tries to locate `index.jsx?` file, in this JS file it looks for the default export.
It then verifies the default export is a function or a react class or React.createClass and adds to it the `withStyles` HOC as well as import for it and import for the `scss` files.

Example:
```diff
--- a/client/reader/update-notice/index.jsx
+++ b/client/reader/update-notice/index.jsx
@@ -14,6 +14,9 @@ import Gridicon from 'gridicons';
 import DocumentHead from 'components/data/document-head';
 import { getDocumentHeadCappedUnreadCount } from 'state/document-head/selectors';
 
+import style from "./_style.scss";
+import withStyles from "isomorphic-style-loader/lib/withStyles";
+
 const UpdateNotice = React.createClass( {
        mixins: [ PureRenderMixin ],
 
@@ -53,4 +56,4 @@ export default connect(
        state => ( {
                cappedUnreadCount: getDocumentHeadCappedUnreadCount( state )
        } )
-)( UpdateNotice );
+)( withStyles(style)(UpdateNotice) );
```

For all those `scss` files successfully processed above,
it checks if the `scss` files are missing imports for mixins or variables and adds them.
For example:
```diff
--- a/client/blocks/edit-gravatar/style.scss
+++ b/client/blocks/edit-gravatar/style.scss
@@ -1,3 +1,5 @@
+@import 'assets/stylesheets/shared/mixins/breakpoints';
+@import 'assets/stylesheets/shared/colors';
 .edit-gravatar__image-container {
        display: inline-block;
        position: relative;
```

**Note:**
SCSS Files that have `@extend %whatever` or `@extend .whatever`, are skipped,
therefor the `jsx?` file is skipped as well.
( because we need to figure out where the .whatever selector is located and import that file )

It also writes the successfully processed list to `client/exclude-scss-from-build`, so in the [Makefile](https://github.com/Automattic/wp-calypso/pull/12210/files#diff-b67911656ef5d18c4ae36cb6741b7965R31) they'll be excluded from the build of `styles.css` :clap: @stephanethomas 

Those scss are also removed from the imports at `assets/stylesheets/_components.scss`
:clap: @drewblaisdell 

### TODO
- [ ] ~Handle scss error:  `@extend %XXXXX` `The selector "%XXXXX" was not found.`~ ignore for now
- [x] Handle scss error: Refactor the variables `$sidebar-width-max` from `assets/stylesheets/main` to another file
- [ ] ~Handle scss error: `./~/sass-loader!./client/my-sites/sharing/style.scss` `Base-level rules cannot contain the parent-selector-referencing character '&'.`~ ignore for now
- [x] `client/my-sites/design-preview/index.js` is not transformed correctly because it's a HOC
- [x] verify whether the exported function returns JSXElement, and not just trust it to be a react component, as `client/reader/index.js`
- [ ] [Refactor](https://github.com/Automattic/wp-calypso/pull/12210/commits/da0971e6beee140bcb461440da13208ffe16e0de) all instances that render root of react tree to use `renderWithReduxStore` helper
- [ ] Handle RTL

### CODE REVIEW

File: server/pages/index.jade [hunk 1]
```diff
@@ -54,6 +54,8 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
 				link(rel='stylesheet', href=urls['style-debug.css'])
 			else
 				link(rel='stylesheet', href=urls['style.css'])
+		style(type='text/css' id='critical-path-css')
```
Comment (scruffian) [inline server/pages/index.jade]: Should we make this change to `server/pages/desktop.jade` as well?

**Comment by scruffian** [comment]
This looks good. I think it would be good to add an example of how it would work for components that use bundled CSS...

**Comment by yurynix** [comment]
Since this PR got large, I'll attempt to split it to smaller PRs.

**Comment by scruffian** [comment]
We decided to take a different approach

**Pull request closed without merging**